File: LeetCode/Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) == 0:
            return 0
        mi = prices[0]
        mx = 0
        for i in range(len(prices)):
            mi = min(mi, prices[i])
            mx = max(mx, prices[i] - mi)
        return mx

File: NowCoder/leetocde/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        int sz = prices.size();
        if(sz == 0) {
            return 0;
        }
        int low = prices[0];
        int ans = 0;
        for(int i = 1; i < sz; ++i) {
            if(prices[i] < low) {
                low = prices[i];
            }else {
                ans = max(prices[i] - low, ans);
            }
        }
        return ans;
    }
};